pkg/serverless/daemon/daemon.go

```diff
@@ -66,6 +66,18 @@ type Daemon struct {
 	// LambdaLibraryDetected represents whether the Datadog Lambda Library was detected in the environment
 	LambdaLibraryDetected bool
 
+	// LambdaLibraryStateLock keeps track of whether the Datadog Lambda Library was detected in the environment
+	LambdaLibraryStateLock sync.RWMutex
```

Comment:
Why are you using a `RWMutex` here but a `Mutex` on the other? In some cases a `RWMutex` can be a lot slower than a `Mutex`. I know I was the one who suggested you use these, but I would recommend running some tests in the wild to make sure you've chosen the fastest mutex for the situation.

Reply:
Can do 👍  I'll run some tests and update this

Reply:
Attempted to use a benchmark and there didn't seem to be a huge difference between using RWMutex vs Mutex. Functions were identical except using RLock/RUnlock vs Lock/Unlock:
```sh
BenchmarkIsLambdaLibraryDetected # Mutex
BenchmarkIsLambdaLibraryDetected-10      	1000000000	        14.81 ns/op
BenchmarkIsLambdaLibraryDetectedRW # RWMutex
BenchmarkIsLambdaLibraryDetectedRW-10    	1000000000	        14.82 ns/op
```
If this is really the case (or did I miss something), is it preferred to use the RWMutex so readers aren't blocked?